euphonic/data/bands.py

```diff
@@ -122,6 +125,20 @@ def _get_data(self, seedname, model, path):
         except KeyError:
             pass
 
+        try:
+            if data['model'].lower() == 'castep':
+                try:
+                    self.seedname = data['seedname']
+                    self.model = data['model']
+                    self.path = data['path']
+                except: #TODO warn
+                    self.seedname = None
+                    self.model = None
+                    self.path = ''
+        except: #TODO warn
+            pass
```

Comment:
Could we change this part to:

```
try:
    self.model = data['model']
    if data['model'].lower() == 'castep':
        self.seedname = data['seedname']
        self.model = data['model']
        self.path = data['path']
except KeyError:
    pass
```

So we'll except on a KeyError and just carry on if the model key isn't there, and for now we'll just assume that if model == 'CASTEP' the seedname, model and path keys are also there, as they should be set by the `_castep._read_interpolation_data` function.

And the same thing for the `_get_data` function in `InterpolationData` and `PhononData`